src/libasr/codegen/asr_to_python.cpp

```diff
@@ -0,0 +1,540 @@
+            case (ASR::cmpopType::LtE) : return " <= ";
+            case (ASR::cmpopType::Gt) : return " > ";
+            case (ASR::cmpopType::GtE) : return " >= ";
+            default : throw LCompilersException("Cannot represent the boolean operator as a string");
+        }
+    }
+
+    std::string logicalbinop2str(const ASR::logicalbinopType type)
+    {
+        switch (type) {
+            case (ASR::logicalbinopType::And) : {
+                last_expr_precedence = Precedence::And;
+                return " and ";
+            } case (ASR::logicalbinopType::Or) : {
+                last_expr_precedence = Precedence::Or;
+                return " or ";
+	    } default : {
+                throw LCompilersException("Cannot represent the boolean operator as a string");
+            }
+        }
+    }
+
+    template <typename T>
+    void visit_body(const T &x, std::string &r, bool apply_indent=true) {
+        if (apply_indent) {
+            inc_indent();
+        }
+        for (size_t i = 0; i < x.n_body; i++) {
+            visit_stmt(*x.m_body[i]);
+            r += s;
+        }
+        if (apply_indent) {
+            dec_indent();
+        }
+    }
+
+    std::string get_type(const ASR::ttype_t *t) {
+        std::string r = "";
+        switch (t->type) {
+            case ASR::ttypeType::Integer : {
+                std::string out;
+                out = std::to_string(down_cast<ASR::Integer_t>(t)->m_kind);
+                if (out == "1") {
+                    r += "i8";
+                } else if (out == "2") {
+                    r += "i16";
+                } else if (out == "4") {
+                    r += "i32";
+                } else if (out == "8") {
+                    r += "i64";
+                }
+                break;
+            } case ASR::ttypeType::Real : {
+                std::string out;
+                out = std::to_string(down_cast<ASR::Real_t>(t)->m_kind);
+                if (out == "4") {
+                    r += "f32";
+                } else if (out == "8") {
+                    r += "f64";
+                }
+                break;
+            } case ASR::ttypeType::Complex : {
+                std::string out;
+                out = std::to_string(down_cast<ASR::Complex_t>(t)->m_kind);
+                if (out == "4") {
+                    r += "c32";
+                } else if (out == "8") {
+                    r += "c64";
+                }
+                break;
+            } case ASR::ttypeType::Character : {
+                r = "str";
+                break;
+            } case ASR::ttypeType::Logical : {
+                r = "bool";
+                break;
+            } case ASR::ttypeType::Array: {
+                r = get_type(down_cast<ASR::Array_t>(t)->m_type);
+                break;
+            } case ASR::ttypeType::Allocatable: {
+                r = get_type(down_cast<ASR::Allocatable_t>(t)->m_type);
+                break;
```

Comment:
```suggestion
            } case ASR::ttypeType::Array: {
                r = get_type(down_cast<ASR::Array_t>(t)->m_type);
                break;
            } case ASR::ttypeType::Allocatable: {
                r = get_type(down_cast<ASR::Allocatable_t>(t)->m_type);
                break;
```
This might not work, as expected. I think we can remove this or implement it and compare the output with the source code.